### PR TITLE
Throw error when using partitioned ODE methods with ODEProblem

### PR DESCRIPTION
**Pull request by ErikQQY**
This PR is a simple fix of the error message for the case when we use the methods for partitioned ODE with ODEProblem.

Fix: #1738 

error

I think maybe we can also define a specific type for these partitioned ODE i.e. `OrdinaryDiffEqPartitionedAlgorithms`, it would be easier for the judgement.

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]

File: src/solve.jl [hunk 1]
```diff
@@ -78,6 +78,18 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
         error("You cannot use an DAE Algorithm with a ODEProblem")
     end
 
+    partitioned_ode_solvers = [Nystrom4(), IRKN3(), IRKN4(), ERKN4(), ERKN5(), ERKN7(), Nystrom4VelocityIndependent(),
+                               Nystrom5VelocityIndependent(), DPRKN4(), DPRKN5(), DPRKN6(), DPRKN6FM(), DPRKN8(), DPRKN12(),
+                               SymplecticEuler(), VelocityVerlet(), VerletLeapfrog(), PseudoVerletLeapfrog(), McAte2(), Ruth3(),
+                               McAte3(), CandyRoz4(), McAte4(), CalvoSanz4(), McAte42(), McAte5(), Yoshida6(), KahanLi6(),
+                               McAte8(), KahanLi8(), SofSpa10()]
```
Comment (ChrisRackauckas) [inline src/solve.jl:85]: Use a subtype instead

File: src/solve.jl [hunk 2]
```diff
@@ -78,6 +78,18 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
...
+
+    if prob isa DiffEqBase.AbstractODEProblem
+        if in(alg, partitioned_ode_solvers)
+            error("You cannot use an algorithm designed for partitioned ODE with an ODEProblem, please use DynamicalODEProblem/SecondOrderODEProblem/HamiltonianProblem instead.")
```
Comment (ChrisRackauckas) [inline src/solve.jl:89]: Those generate an ODEProblem though. It should instead check the `prob.problem_type`.

File: src/solve.jl (hunk 1 above)
Comment (ErikQQY) [reply src/solve.jl:85]: I am not sure if I am getting it right, so we need to specify a type of algorithms like ```abstract OrdinaryDiffEqPartitionedAlgorithms <: OrdinaryDiffEqAlgorithms end```, and then redefine all of the partitioned ODE methods like ```struct Nystrom4 <: OrdinaryDiffEqPartitionedAlgorithms end```?

Comment (ChrisRackauckas) [reply src/solve.jl:85]: Yes, otherwise that type check allocates and is not inferred.

**Comment by ErikQQY** [comment]
odewithrkn
odewithtsit5

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Looks like tests are failing.

**Pull request merged**